src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +135,1023 @@ static DamageTable D_80A63CE0 = {
-        { 6, { { 2400, 0, 0 }, 50 }, 100 },
+        {
+            ELEMTYPE_UNK0,
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            { 0xF7CFFFFF, 0x00, 0x00 },
+            TOUCH_NONE | TOUCH_SFX_NORMAL,
+            BUMP_ON,
+            OCELEM_ON,
+        },
+        { TINGLE_LIMB_BALLOON, { { 2400, 0, 0 }, 50 }, 100 },
     },
 };
 
-// static ColliderJntSphInit sJntSphInit = {
-static ColliderJntSphInit D_80A63D30 = {
-    { COLTYPE_NONE, AT_ON | AT_TYPE_ENEMY, AC_ON | AC_TYPE_PLAYER, OC1_ON | OC1_TYPE_ALL, OC2_TYPE_1, COLSHAPE_JNTSPH, },
-    ARRAY_COUNT(sJntSphElementsInit), D_80A63D0C, // sJntSphElementsInit,
+static ColliderJntSphInit sJntSphInit = {
+    {
+        COLTYPE_NONE,
+        AT_ON | AT_TYPE_ENEMY,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_JNTSPH,
+    },
+    ARRAY_COUNT(sJntSphElementsInit),
+    sJntSphElementsInit,
+};
+
+static s16 sBuyMapOptions[TINGLE_MAP_MAX][2] = {
+    { TINGLE_MAP_CLOCK_TOWN, TINGLE_MAP_WOODFALL },   { TINGLE_MAP_WOODFALL, TINGLE_MAP_SNOWHEAD },
+    { TINGLE_MAP_SNOWHEAD, TINGLE_MAP_ROMANI_RANCH }, { TINGLE_MAP_ROMANI_RANCH, TINGLE_MAP_GREAT_BAY },
+    { TINGLE_MAP_GREAT_BAY, TINGLE_MAP_STONE_TOWER }, { TINGLE_MAP_STONE_TOWER, TINGLE_MAP_CLOCK_TOWN },
+};
+
+static AnimationInfo sAnimationInfo[TINGLE_ANIM_MAX] = {
+    { &gTingleFloatIdleAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },     // TINGLE_ANIM_FLOAT_DRAW
+    { &gTingleFallAnim, 1.5f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },          // TINGLE_ANIM_FALL_LOOP
+    { &gTingleFallAnim, 1.5f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f },          // TINGLE_ANIM_FALL_ONCE
+    { &gTingleLandAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -4.0f },          // TINGLE_ANIM_LAND
+    { &gTingleTalkTwistAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },     // TINGLE_ANIM_TALK_TWIST
+    { &gTingleTalkIdleAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },      // TINGLE_ANIM_TALK_IDLE
+    { &gTingleMagicSpellAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -8.0f },    // TINGLE_ANIM_MAGIC_FULL
+    { &gTingleHappyDanceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },    // TINGLE_ANIM_HAPPY_DANCE_LOOP
+    { &gTingleHappyDanceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_ONCE, -8.0f },    // TINGLE_ANIM_HAPPY_DANCE_ONCE
+    { &gTingleMagicSpellAnim, 1.0f, 23.0f, 0.0f, ANIMMODE_ONCE, -8.0f },   // TINGLE_ANIM_MAGIC_SHORT
+    { &gTingleWaitAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -8.0f },          // TINGLE_ANIM_WAIT
+    { &gTingleSpinAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },          // TINGLE_ANIM_SPIN
+    { &gTingleHideFaceAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f },      // TINGLE_ANIM_HIDE_FACE
+    { &gTingleThrowConfettiAnim, 1.0f, 0.0f, 0.0f, ANIMMODE_LOOP, -2.0f }, // TINGLE_ANIM_CONFETTI
 };
 
-#endif
+void EnBal_Init(Actor* thisx, PlayState* play) {
+    EnBal* this = THIS;
+    s32 pad;
+    f32 lastFrame = Animation_GetLastFrame(&gTingleFloatIdleAnim);
```

Comment:
```suggestion
    f32 endFrame = Animation_GetLastFrame(&gTingleFloatIdleAnim);
```
I know the function is called `Animation_GetLastFrame`, but we've been transitioning to using `endFrame` and I will likely propose to change `Animation_GetLastFrame` soon